kong/pdk/log.lua

```diff
@@ -465,11 +474,158 @@ local _log_mt = {
 }
 
 
-local serialize
+local function get_default_serialize_values()
+  if ngx.config.subsystem == "http" then
+    return {
+      { key = "request.headers.authorization", value = "REDACTED", mode = "replace" },
+      { key = "request.headers.proxy-authorization", value = "REDACTED", mode = "replace" },
+    }
+  end
+
+  return {}
+end
+
+---
+-- Sets a value to be used on the `serialize` custom table
+--
+-- Logging plugins use the output of `kong.log.serialize()` as a base for their logs.
+--
+-- This function allows customizing such output.
+--
+-- It can be used to replace existing values on the output.
+-- It can be used to delete existing values by passing `nil`.
+--
+-- Note: the type checking of the `value` parameter can take some time so
+-- it is deferred to the `serialize()` call, which happens in the log
+-- phase in most real-usage cases.
+--
+-- @function kong.log.set_serialize_value
+-- @phases certificate, rewrite, access, header_filter, body_filter, log
+-- @tparam string key the name of the field.
+-- @tparam number|string|boolean|table value value to be set. When a table is used, its keys must be numbers, strings, booleans, and its values can be numbers, strings or other tables like itself, recursively.
```

Comment:
I suppose arrays are supported? And what kind of arrays? Even arrays with hash parts? What about `ngx.null` AND `ngx.null` handling in general in this code (perhaps not support `ngx.nulls`, but can there be `ngx.nulls` already in existing log data)?

Reply:
Arrays are supported, but usually we will see them on the `value` field - `kong.log.set_serialize_value("foo", {1,2,3})`. Values can be arrays or hybrid tables.

We should not see it on the `key` field because the table returned by `kong.log.serialize()` is a hash-like array.

`ngx.null` is not permitted in `set_serialize_value`, but it is permitted in existing log data.